elastic/kibana
# [Stack Monitoring] support for integration package

### Summary

**This change should be tested with https://github.com/elastic/integrations/pull/3965**

Two updates required to support the kibana integration package:
- added a metricset filter to the kibana instance overview query. this is required because two document types (stats and status) can be matched with the current filters but only stats has the right properties used by the UI. With metricbeat standalone this is not a problem because we only collect the [stats metricset](https://github.com/elastic/beats/blob/main/metricbeat/module/kibana/kibana.go#L93) but with integrations both metricsets can be collected
- there is a guard to only display the Rules/Actions data if the documents come from the metricbeat indice. Extended the condition to also allow metrics data streams

### Testing
- first setup an elastic-package as described in https://github.com/elastic/integrations/pull/3965
- update kibana_system password on the elastic-package's elasticsearch
``` 
curl --insecure -u elastic:changeme -H 'Content-Type: application/json' \
https://localhost:9200/_security/user/kibana_system/_password -d'{"password": "changeme"}'
```
- configure kibana on a different port so we don't conflict with the one started by elastic-package and ignore certificate verification
```
server.port: 5602

elasticsearch.username: kibana_system
elasticsearch.password: changeme
elasticsearch.ignoreVersionMismatch: true
elasticsearch.hosts: ["https://localhost:9200"]
elasticsearch.ssl.verificationMode: none
```
- connect to local kibana at `localhost:5602`. Stack Monitoring should already have elasticsearch and kibana data, verify all kibana views are correctly populated

------

Note that the local kibana is not able to interact with the local package-registry when running http, this means we have to do all integration-related interaction on the kibana run by elastic-package (at :5601).
To enable ssl we can point the local kibana configuration to the certificates used by elastic-package (under `~/.elastic-package/profiles/{profile}/certs`), then start the process with `NODE_EXTRA_CA_CERTS=~/.elastic-package/profiles/{profile}/certs/ca-cert.pem yarn start`. With that everything appears to work well except that any successful update to an agent policy from the local kibana is not picked up by the Agent, only the elastic-package updates go through. I didn't find the cause yet
